### PR TITLE
test: add e2e for conjur csi provider

### PR DESCRIPTION
**Pull request by gl-johnson**
**What type of PR is this?**

**What this PR does / why we need it**:
Adds e2e test retrieving secrets via [the Conjur provider](https://docs.cyberark.com/conjur-enterprise/latest/en/Content/Integrations/k8s-ocp/k8s-jwt-secrets-store-csi-driver.htm)

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [x] includes documentation (Follow-up PR #1521)
- [ ] adds unit tests

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
/ok-to-test

**Comment by gl-johnson** [comment]
Hi @aramase @nilekhc - any chance this can be reviewed this week? Let me know if you have any feedback/questions. Thanks!

File: test/bats/conjur.bats [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+EXPECTED_PASSWORD="SecretPassword1234!"
+
+@test "allow conjur provider token requests and enable rotation" {
+  helm upgrade csi-secrets-store manifest_staging/charts/secrets-store-csi-driver --namespace=kube-system \
+    --set tokenRequests[0].audience="conjur" \
+		--set enableSecretRotation=true \
+		--set rotationPollInterval=30s
+  
+  kubectl wait --for=condition=ready --timeout=${WAIT_TIME}s --namespace=kube-system pod -l "app=secrets-store-csi-driver"
+}
```
Comment (aramase) [inline test/bats/conjur.bats:24]: Could you add `--set tokenRequests[2].audience="conjur"` to https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/5f6daa036572b1cda059c20ecefb2ad4a642e6f8/Makefile#L439-L440 instead and remove this step?

File: test/bats/conjur.bats [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+@test "install conjur provider" { 
+  # Update Helm repos
+  helm repo add cyberark https://cyberark.github.io/helm-charts || true
+  helm repo update
+
+  # Create Conjur namespace
+  kubectl create namespace $CONJUR_NAMESPACE
```
Comment (aramase) [inline test/bats/conjur.bats:32]: You can use `--create-namespace` flag as part of `helm install` command

File: test/bats/conjur.bats [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+  # Install Conjur
+  helm install conjur cyberark/conjur-oss --namespace=$CONJUR_NAMESPACE --wait --timeout ${WAIT_TIME}s \
+    --set dataKey=$CONJUR_DATA_KEY \
+    --set authenticators="authn\,authn-jwt/kube" \
+    --set service.external.enabled=false
+
+  # Install Conjur CSI Provider
+  helm install conjur-csi-provider cyberark/conjur-k8s-csi-provider --wait --timeout ${WAIT_TIME}s --namespace=kube-system \
+    --set providerServer.image.tag=latest
```
Comment (aramase) [inline test/bats/conjur.bats:42]: `latest` tag in provider could break CI in the driver. Is there a pinned version you can use here?

File: test/bats/conjur.bats [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+@test "setup conjur" { 
+  # Create Conjur account and store admin API key
+  admin_api_key="$(kubectl exec deployment/conjur-conjur-oss \
+    --namespace=$CONJUR_NAMESPACE \
+    --container=conjur-oss \
+    -- conjurctl account create $CONJUR_ACCOUNT | grep API | awk '{print $5}')"
+
+  # Create a Conjur CLI pod
+  docker pull cyberark/conjur-cli:8
+  kind load docker-image cyberark/conjur-cli:8
```
Comment (aramase) [inline test/bats/conjur.bats:54]: The image is public, so I don't think this step is required.

File: test/bats/conjur.bats [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  envsubst < $BATS_TESTS_DIR/conjur_v1_secretproviderclass.yaml | sed '/^ *-----BEGIN CERTIFICATE-----/,$s/^/      /' | kubectl apply --namespace=kube-system -f -
+
+  kubectl wait --for condition=established --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/conjur -o yaml --namespace=kube-system | grep conjur"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+}
+
+@test "CSI inline volume test with pod portability" {
+  kubectl apply --namespace=kube-system -f $BATS_TESTS_DIR/pod-secrets-store-inline-volume-crd.yaml
```
Comment (aramase) [inline test/bats/conjur.bats:97]: Does the workload need to be in `kube-system` namespace? If someone looks at this file for reference, I don't want them to assume this is a requirement. Could we use any namespace other than `kube-system` for the workloads being deployed?

**Review by aramase — CHANGES_REQUESTED** [review]

File: test/bats/conjur.bats (hunk 5 above)
Comment (gl-johnson) [reply test/bats/conjur.bats]: I've moved it test workloads to the `default` namespace

File: test/bats/conjur.bats [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+EXPECTED_PASSWORD="SecretPassword1234!"
+
+@test "install conjur provider" { 
+  # Update Helm repos
+  helm repo add cyberark https://cyberark.github.io/helm-charts || true
+  helm repo update
+
+  # Install Conjur
+  helm install conjur cyberark/conjur-oss --namespace=$CONJUR_NAMESPACE --wait --timeout ${WAIT_TIME}s \
+    --create-namespace --namespace=$CONJUR_NAMESPACE \
```
Comment (aramase) [inline test/bats/conjur.bats:24]:
```suggestion
  helm install conjur cyberark/conjur-oss --namespace=$CONJUR_NAMESPACE --wait --timeout ${WAIT_TIME}s \
    --create-namespace  \
```

`--namespace` flag is already defined

**Review by aramase — COMMENTED** [review]
Just one minor comment!

After the update, could you squash to a single commit?

File: test/bats/conjur.bats (hunk 6 above)
Comment (gl-johnson) [reply test/bats/conjur.bats]: Woops - all should be addressed now!

**Review by aramase — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**